core/src/main/java/hudson/PluginManager.java

```diff
@@ -1923,6 +1924,32 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
         return FormValidation.ok();
     }
 
+    @Restricted(NoExternalUse.class)
+    @RequirePOST public FormValidation doCheckUpdateSiteUrl(StaplerRequest request, @QueryParameter String value) {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        if (StringUtils.isNotBlank(value)) {
+            try {
+                value += ((value.contains("?")) ? "&" : "?") + "version=" + Jenkins.VERSION + "&uctest";
+
+                URL url = new URL(value);
+
+                // Connect to the URL
+                HttpURLConnection conn = (HttpURLConnection) ProxyConfiguration.open(url);
+                conn.setRequestMethod("HEAD");
+                conn.setConnectTimeout(5000);
```

Comment:
should this also have a read timeout?

```suggestion
                conn.setReadTimeout(1000);
```

Reply:
Will 1 sec too short? How about 5 sec which is same as connect timeout?